File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/EpisodeOfCare.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class EpisodeOfCare : DomainResource, IResource
{
    public required Base.ResourceReference Patient { get; set; }
    public EpisodeOfCare_Diagnosis[]? Diagnosis { get; set; }
    public Base.ResourceReference? ManagingOrganization { get; set; }
    public Base.CodeableConcept[]? Type { get; set; }
    public Base.ResourceReference[]? Account { get; set; }
    public Base.ResourceReference[]? ReferralRequest { get; set; }
    public Base.ResourceReference[]? Team { get; set; }
    public required string Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.Period? Period { get; set; }
    public Base.ResourceReference? CareManager { get; set; }
    public EpisodeOfCare_StatusHistory[]? StatusHistory { get; set; }

    public class EpisodeOfCare_Diagnosis : BackboneElement
    {
        public string? Rank { get; set; }
        public Base.CodeableConcept? Role { get; set; }
        public required Base.ResourceReference Condition { get; set; }
    }

    public class EpisodeOfCare_StatusHistory : BackboneElement
    {
        public required Base.Period Period { get; set; }
        public required string Status { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/EpisodeOfCare.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class EpisodeOfCareDiagnosis(BackboneElement):
    rank: Optional[int] = None
    role: Optional[CodeableConcept] = None
    condition: Reference

@dataclass(kw_only=True)
class EpisodeOfCareStatusHistory(BackboneElement):
    period: Period
    status: str

@dataclass(kw_only=True)
class EpisodeOfCare(DomainResource):
    patient: Reference
    diagnosis: Optional[List[EpisodeOfCareDiagnosis]] = field(default_factory=list)
    managing_organization: Optional[Reference] = None
    type: Optional[List[CodeableConcept]] = field(default_factory=list)
    account: Optional[List[Reference]] = field(default_factory=list)
    referral_request: Optional[List[Reference]] = field(default_factory=list)
    team: Optional[List[Reference]] = field(default_factory=list)
    status: str
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    period: Optional[Period] = None
    care_manager: Optional[Reference] = None
    status_history: Optional[List[EpisodeOfCareStatusHistory]] = field(default_factory=list)